### ISSUE TITLE
about shadow_model train datasets

### ISSUE THREAD

**Issue by wpyniub** [icmpnorequest/MLSec#2]
Hallo,i have a question about generating shadow model training data.
you are using a query-based synthesis approach.However, the display result of the synthesis data is far from that of the cifar-10 picture.
Can you tell me why?Thank you very much！

**Comment by icmpnorequest** [comment]
> Hallo,i have a question about generating shadow model training data.
> you are using a query-based synthesis approach.However, the display result of the synthesis data is far from that of the cifar-10 picture.
> Can you tell me why?Thank you very much！

Hi,

The code of this paper doesn't contain data synthesis algorithm, so I just completed the algorithm on my own. I have contacted Dr. Shokri, the first author of the paper, but he told me the algorithm is designed for binary class. In other words, it doesn't fit for Cifar10 dataset. Therefore, we need to complete the data synthesis algorithm for Cifar10 dataset by ourselves.

**Comment by wpyniub** [comment]
Thank you very much for answering my question. One question is why the accuracy of the attack in your code is very high when the cifar-10 synthesis data obtained by using the query is quite different from the real data.And do you know what methods are available to synthesize simulated image information in the black box situation.

**Comment by icmpnorequest** [comment]
> Thank you very much for answering my question. One question is why the accuracy of the attack in your code is very high when the cifar-10 synthesis data obtained by using the query is quite different from the real data.And do you know what methods are available to synthesize simulated image information in the black box situation.

The original code for the paper just replace some images with some random generated numbers, so I guess the key intuition lies on shadow models generation and shadow training (depends heavily on the guessing of shadow model). That might be why the accuracy of membership inference attack is high. For the second question, maybe you could refer to membership inference attack and adversarial example generation.

**Comment by wpyniub** [comment]
好滴，学长你的意思是攻击的精度与影子训练数据和影子模型直接相关吧。然后这个是从合成的数据集中选取一条数据生成的图像，是我生成的方式错误了吗，咋是这个结果。谢谢学长啦





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "icmpnorequest/MLSec"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年10月7日(星期三) 下午3:10
收件人:&nbsp;"icmpnorequest/MLSec"<MLSec@noreply.github.com&gt;;
抄送:&nbsp;"王先森"<517928277@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [icmpnorequest/MLSec] about shadow_model train datasets (#2)





  
Thank you very much for answering my question. One question is why the accuracy of the attack in your code is very high when the cifar-10 synthesis data obtained by using the query is quite different from the real data.And do you know what methods are available to synthesize simulated image information in the black box situation.
  
The original code for the paper just replace some images with some random generated numbers, so I guess the key intuition lies on shadow models generation and shadow training (depends heavily on the guessing of shadow model). That might be why the accuracy of membership inference attack is high. For the second question, maybe you could refer to membership inference attack and adversarial example generation.
